### PR TITLE
.Net Restaurant Bookings Demo

### LINKED ISSUE #5960: .Net Demo (Booking Agent Sample)
### Integrations

- Microsoft Bookings API (MSGraph)
- Semantic Kernel - BookPlugin
 
### Scenario 

- The User wants to book a reservation at a restaurant through use of the AI agent. They want the booking to be at a time the restaurant is available for 2 individuals and past 6pm on a weekday. 

- The User will need the ability to modify the reservation if needed for their schedule to a different free slot at the restaurant. 

- The User will need the ability to cancel the reservation.  

- The User wants an autonomous agent to schedule reservations at their favorite restaurant given their schedule constraints.

### Actions

- Ensure Agent Ask to the User before it confirms the reservation.

### PR DESCRIPTION
**Pull request by RogerBarreto**
### Motivation and Context

Resolves #5960 

This pull request includes several changes to the `dotnet` directory, mainly adding a new project called `BookingRestaurant` and updating dependencies. The `BookingRestaurant` is a demo application for booking restaurant tables using Microsoft Graph Bookings API. The most important changes include adding a new project to the solution, updating package versions, and adding new patterns to the markdown link check config.

New Project:

* [`dotnet/SK-dotnet.sln`](diffhunk://#diff-6728f39c7d31a256668698ee869487799bf43c93dc8cc7e889befe95b0ec09e6R262): Added a new project called `BookingRestaurant` to the solution. [[1]](diffhunk://#diff-6728f39c7d31a256668698ee869487799bf43c93dc8cc7e889befe95b0ec09e6R262) [[2]](diffhunk://#diff-6728f39c7d31a256668698ee869487799bf43c93dc8cc7e889befe95b0ec09e6R623-R628) [[3]](diffhunk://#diff-6728f39c7d31a256668698ee869487799bf43c93dc8cc7e889befe95b0ec09e6R717)
* [`dotnet/samples/Demos/BookingRestaurant/*`](diffhunk://#diff-f846bffbf2145055d4ab6e138b8e20d2ada4000292541e98d5b242c250327881R1-R74): Added several new files for the `BookingRestaurant` project, including `AppConfig.cs`, `Appointment.cs`, `BookingRestaurant.csproj`, `BookingsPlugin.cs`, and `Program.cs`. These files contain the main logic and configuration for the new project. [[1]](diffhunk://#diff-f846bffbf2145055d4ab6e138b8e20d2ada4000292541e98d5b242c250327881R1-R74) [[2]](diffhunk://#diff-5b3c66c05657fed5958d03e136b3c2376aaa1d9afea5d30c0653e69c92e39890R1-R24) [[3]](diffhunk://#diff-ff4fe71831c229af576b18e123e87869077b64f9d41338f02fe583e35b1f761bR1-R30) [[4]](diffhunk://#diff-21617b4ad8043159f5d5d0f394e839bc3b700e7bf965cd75b037fb0149e0c3cbR1-R140) [[5]](diffhunk://#diff-624c47bcfbbff8c6156f7056d410a92b05e0a0c4038172b213903e03fa81511eR1-R128)

### CODE REVIEW

File: dotnet/samples/Demos/BookingRestaurant/AppConfig.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    internal sealed class AzureOpenAIConfig
+    {
+        public string? DeploymentName { get; set; }
+        public string? Endpoint { get; set; }
+        public string? ApiKey { get; set; }
+    }
+
+    internal sealed class AzureAdConfig
```
Comment (markwallace-microsoft) [inline dotnet/samples/Demos/BookingRestaurant/AppConfig.cs:66]: Update to `EntraIdConfig`

File: dotnet/samples/Demos/BookingRestaurant/AppConfig.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            ArgumentNullException.ThrowIfNull(this.AzureAd?.ClientSecret, nameof(this.AzureAd.ClientSecret));
+        }
+    }
+
+    internal sealed class OpenAIConfig
+    {
+        /// <summary>
+        /// The model ID to use for the OpenAI chat completion.
+        /// Available Chat Completion models can be found at https://platform.openai.com/docs/models.
+        /// </summary>
```
Comment (markwallace-microsoft) [inline dotnet/samples/Demos/BookingRestaurant/AppConfig.cs:53]: Any reason to just document this property

File: dotnet/samples/Demos/BookingRestaurant/BookingsPlugin.cs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        this._graphClient = graphClient;
+        this._businessId = businessId;
+        this._serviceId = serviceId;
+        this._customerTimeZone = customerTimeZone;
+    }
+
+    [KernelFunction("BookTable")]
+    [Description("Books a new table at a restaurant")]
+    public async Task<string> BookTableAsync(
+        string restaurant,
```
Comment (markwallace-microsoft) [inline dotnet/samples/Demos/BookingRestaurant/BookingsPlugin.cs:36]: Add description for all parameters as this is the pattern we want to encourage

File: dotnet/samples/Demos/BookingRestaurant/BookingsPlugin.cs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                {
+                    OdataType = "#microsoft.graph.dateTimeTimeZone",
+                    DateTime = dateTime.AddHours(2).ToString("o"),
+                    TimeZone = "UTC",
+                },
+                IsLocationOnline = false,
+                OptOutOfCustomerEmail = false,
+                AnonymousJoinWebUrl = null,
+                PostBuffer = TimeSpan.FromMinutes(10),
+                PreBuffer = TimeSpan.FromMinutes(5),
```
Comment (markwallace-microsoft) [inline dotnet/samples/Demos/BookingRestaurant/BookingsPlugin.cs:65]: Define constants for the magic numbers

**Review by markwallace-microsoft — APPROVED** [review]

**Review by matthewbolanos — APPROVED** [review]

**Review by matthewbolanos — APPROVED** [review]

**Review by sophialagerkranspandey — APPROVED** [review]

**Pull request merged**